lib/any.js

```diff
@@ -519,7 +521,7 @@ internals.Any.prototype._validate = function (value, state, options, reference)
                     errors.push(Errors.create('any.default', defaultValue.error, state, options));
                 }
             }
-            else {
+            else if (!options.noDefaults) {
```

Comment:
This is not enough, you're only short-circuiting the non-function form of default (previous if).